src/time.js

```diff
@@ -186,7 +186,7 @@ function _convertItemRawState (rawState) {
   } else if (rawState instanceof StringType) { // String type Items
     return _parseString(rawState.toString());
   } else if (rawState instanceof DateTimeType) { // DateTime Items
-    return javaZDTToJsZDT(rawState.getZonedDateTime());
+    return javaInstantToJsInstant(rawState.getInstant()).atZone(time.ZoneId.systemDefault());
```

Comment:
I would rather get the ZonedDateTime from the DateTimeType with Java code and convert that Java ZDT to a JS ZDT.

Reply:
You cannot get it from the `DateTimeType` without breaking compatibility with OH 4.3 - see https://github.com/openhab/openhab-js/issues/404#issuecomment-2536814538. But you can of course get it from the Java `Instant` class in the same way as `DateTimeType` does it:

```java
    public ZonedDateTime getZonedDateTime(ZoneId zoneId) {
        return instant.atZone(zoneId);
    }
```

So in this case we would need to:
- Convert `time.ZoneId.systemDefault()` to Java `ZoneId`.
- Call Java `rawState.getInstant.atZone`.
- Convert the result to JS `ZonedDateTime`.

What would be the benefit of doing that rather than doing it directly in JS?

Reply:
Possibly performance impact, but I guess it doesn’t matter really matter given how the javaZDTToJsZDT method works.